From 3168d423d03add8d3fd371e9e98dc71a3cbf4edc Mon Sep 17 00:00:00 2001
From: Loc Vu <loc.vu.zn@renesas.com>
Date: Thu, 25 Jul 2019 17:11:03 +0700
Subject: [PATCH 032/433] clk: shmobile: add support for r8a77470

The previous commit (4be22c15) removed commpatible defined for r8a77470
This commit add it again, and prevent conflict with other SoCs

Signed-off-by: Binh Nguyen <binh.nguyen.uw@rvc.renesas.com>
Signed-off-by: Loc Vu <loc.vu.zn@renesas.com>
---
 drivers/clk/shmobile/clk-rcar-gen2.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/clk/shmobile/clk-rcar-gen2.c b/drivers/clk/shmobile/clk-rcar-gen2.c
index 3f7d331..671a16d 100644
--- a/drivers/clk/shmobile/clk-rcar-gen2.c
+++ b/drivers/clk/shmobile/clk-rcar-gen2.c
@@ -373,7 +373,10 @@ rcar_gen2_cpg_register_clock(struct device_node *np, struct rcar_gen2_cpg *cpg,
 		mult = config->pll3_mult;
 	} else if (!strcmp(name, "lb")) {
 		parent_name = "pll1";
-		div = cpg_mode & BIT(18) ? 36 : 24;
+		if (of_device_is_compatible(np, r8a77470_compat))
+			div = 12;
+		else
+			div = cpg_mode & BIT(18) ? 36 : 24;
 	} else if (!strcmp(name, "qspi")) {
 		parent_name = "pll1_div2";
 		div = (cpg_mode & (BIT(3) | BIT(2) | BIT(1))) == BIT(2)
@@ -470,7 +473,8 @@ static void __init rcar_gen2_cpg_clocks_init(struct device_node *np)
 
 	of_clk_add_provider(np, of_clk_src_onecell_get, &cpg->data);
 
-	cpg_mstp_add_clk_domain(np);
+	if (!of_device_is_compatible(np, r8a77470_compat))
+		cpg_mstp_add_clk_domain(np);
 }
 CLK_OF_DECLARE(rcar_gen2_cpg_clks, "renesas,rcar-gen2-cpg-clocks",
 	       rcar_gen2_cpg_clocks_init);
-- 
2.7.4

